templates/default/condasearch.sh.erb

```diff
@@ -1,15 +1,17 @@
 #!/bin/bash
 
-# conda search -c $1 $2
-
-if [ $# -ne 2  ]; then
- echo "incorrect usage"
+if [ $# -ne 3  ]; then
+ echo "Usage: $(basename $0) CHANNEL LIBRARY BASE_IMAGE_NAME"
  exit 2
 fi
+
 CHANNEL="${1//[^a-zA-Z0-9\.\-\_:\/\&~?]/}"
 if [ "$CHANNEL" != "defaults" ] ; then
    CHANNEL="$CHANNEL --override-channels"
 fi
+
 LIB="${2//[^a-zA-Z0-9\.\-\_]/}"
-<%= node['conda']['base_dir'] %>/bin/conda search -c $CHANNEL "${LIB}"  | grep -v Fetching | awk '{print $1","$2}' | grep -v "^\*"
+
+BASE_IMAGE_NAME="$3"
+docker run --rm $BASE_IMAGE_NAME conda search -c $CHANNEL "${LIB}" | grep -v Fetching | awk '{print $1","$2}' | grep -v "^\*"
```

Comment:
For other conda operations we mount the condarc: https://github.com/logicalclocks/conda-chef/blob/master/templates/default/condarc.erb

I.e when installing a library https://github.com/logicalclocks/hopsworks-ee/blob/5ff93f8336243c1ec1d858902c53ed182a85c045/hopsworks-common/src/main/java/io/hops/hopsworks/common/python/environment/DockerFileController.java#L170

I'm not sure if we also need to do it for search, maybe it was done automatically before when the script was running on the host where the config file was.

Reply:
I believe you're right. We need to use the condarc in under `glassfish` user because it has configuration for more Channels. There are some properties which are currently wrongly templated, I'll have a look.